src/data/equiv/basic.lean

```diff
@@ -1668,21 +1668,34 @@ begin
   simp [equiv.set.image, equiv.set.image_of_inj_on, hf.eq_iff, this],
 end
 
+/-- If `f` has a left-inverse when `α` is nonempty, then `α` is computably equivalent to the range
+of `f`.
+
+While awkward, the `nonempty α` hypothesis on `f_inv` and `hf` allows this to be used when `α` is
+empty too. This hypothesis is absent on analogous definitions on stronger `equiv`s like
+`linear_equiv.of_left_inverse` and `ring_equiv.of_left_inverse` as their typeclass assumptions
+are already sufficient to ensure non-emptiness. -/
```

Comment:
I don't understand how this is easier to use than just having `nonempty α` as a separate hypothesis.

Reply:
Because if `nonempty` were a separate hypothesis, you couldn't instantiate this for `α = fin 0`

Reply:
Okay, but couldn't we just have a separate lemma for that case, without the `f_inv` hypothesis? My point is that it seems like more human effort to set up the arguments to this function, than to decide which of two different lemmas to apply.

Reply:
If the definition (not lemma) weren't set up in the way I wouldn't be able to use it to implement `equiv.set.range` below

Reply:
Would adding a weaker primed variant `range_of_left_inverse'` that requires `[nonempty _]` address your concern?

Reply:
I agree with Scott that the `nonempty` hypothesis is kind of awkward, since the user probably won't use the data in the `nonempty`. So a `range_of_left_inverse'` that just doesn't mention `nonempty` should be good.

Reply:
I've updated with a new `range_of_left_inverse'` with no mention of `nonempty` besides the docstring.